pkg/operator/operator.go

```diff
@@ -774,6 +774,12 @@ func (o *Operator) sync(ctx context.Context, key string) error {
 				newTaskSpec("ConfigurationSharing", tasks.NewConfigSharingTask(o.client, factory, config)),
 			},
 		),
+		// Peripheral tasks are volatile and run after critical tasks to ensure a stable stack rollout.
+		tasks.NewTaskGroup(
+			[]*tasks.TaskSpec{
+				newTaskSpec("Peripherals", tasks.NewPeripheralsTask(o.client, factory)),
+			},
+		),
```

Comment:
Not sure we need a new TaskGroup here. Since all tasks in a TaskGroup will run concurrently, the new task can be added in the existing last task group.
Also the new task should be focused in what it does, check the user provided metrics-deny list and apply it. If we get more tasks like that, they will get their own task implementation. How about calling the task something like `UserKSMDenyListTask`?

Reply:
ACK, I had this in a separate task group in order for all peripherals to execute sequentially after the last one, but we don't need to do that since running that concurrently in the shared config task should achieve the same effect. I've made the change and renamed the file.